test/metric_learn_test.py

```diff
@@ -86,6 +87,16 @@ def test_iris(self, basis):
     csep = class_separation(scml.transform(X), y)
     assert csep < 0.24
 
+  def test_big_n_features(self):
+    X, y = make_classification(n_samples=100, n_classes=3, n_features=60,
+                               n_informative=60, n_redundant=0, n_repeated=0,
+                               random_state=42)
```

Comment:
While making this test, I tried for `n_informative = 45` and `n_redundant = 15 `. But got as a result an empty `components` matrix, because the objective function never went below 1 which is the value it takes on `iter==1`, so the weights stayed with the initialization value. 

Watching at the change in the weights it seems like it is learning ok, just the values for the objective function are bigger (reaches a minimum of 2 or so) while in general values below 1 are achieved very fast.

I haven't looked deeply into it yet, but it seems like a situation that should be addressed.

Reply:
Testing class separation is nice (although it is not good here probably due to dimension compared to number of data points), but I mostly meant making a test to check that the number of bases one obtains is as expected (check that the returned `n_basis` is equal to the expected one, and that the shape of the returned `basis` is also what we expect), parameterized for a few values of `n_samples`, `n_features` and `n_classes`

(of course you should keep the two tests with toy data where you also know exactly what the basis should be, which is nice)

Reply:
I actually thought this test only to test the `n_features > 50` case!

Reply:
OK, then you could keep it and just extend it to also check the number of basis and basis shape, and parametrize it to cover a few values of the above mentioned parameters

Reply:
I was already on it! Just added it in the last commit.

Reply:
Perfect!
For `n_samples`, maybe remove 1000 to avoid slowing down too much the test suite?